File: 프로그래머스/LV2/구명보트.py
from collections import deque
def solution(people, limit):
    answer = 0
    people = deque(sorted(people, reverse=True))
    while people:
        front = people.popleft()
        if front <= limit//2:
            answer += len(people) // 2 + 1
            break
        else:
            answer += 1
            if people and front + people[-1] <= limit:
                people.pop()
    return answer

File: 프로그래머스/LV2/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int i = 0, j = people.length-1;
        for(; i < j; j--){
            if(people[j] <= limit/2) {
                i += (j-i+1) / 2;
                break;
            }
            if(people[i]+people[j] <= limit) i++;
        }
        return people.length - i;
    }
}
